simpleclient/src/main/java/io/prometheus/client/Summary.java

```diff
@@ -38,133 +36,208 @@
+         */
+        public Value get() {
+            Value v = new Value();
+            v.count = count.sum();
+            v.sum = sum.sum();
+            return v;
+        }
+
     }
+
+    // Convenience methods.
+
+
+    /**
+     * Observe the given amount on the summary with no labels.
+     */
+    public void observe(double amt) {
+        noLabelsChild.observe(amt);
+    }
+
     /**
-     * Get the value of the Summary.
+     * Start a timer to track a duration on the summary with no labels.
      * <p>
-     * <em>Warning:</em> The definition of {@link Value} is subject to change.
+     * Call {@link Timer#observeDuration} at the end of what you want to measure the duration of.
      */
-    public Value get() {
-      Value v = new Value();
-      v.count = count.sum();
-      v.sum = sum.sum();
-      return v;
+    public Timer startTimer() {
+        return noLabelsChild.startTimer();
     }
-  }
-
-  // Convenience methods.
-  /**
-   * Observe the given amount on the summary with no labels.
-   */
-  public void observe(double amt) {
-    noLabelsChild.observe(amt);
-  }
-  /**
-   * Start a timer to track a duration on the summary with no labels.
-   * <p>
-   * Call {@link Timer#observeDuration} at the end of what you want to measure the duration of.
-   */
-  public Timer startTimer() {
-    return noLabelsChild.startTimer();
-  }
-
-  @Override
-  public List<MetricFamilySamples> collect() {
-    List<MetricFamilySamples.Sample> samples = new ArrayList<MetricFamilySamples.Sample>();
-    for(Map.Entry<List<String>, Child> c: children.entrySet()) {
-      Child.Value v = c.getValue().get();
-      samples.add(new MetricFamilySamples.Sample(fullname + "_count", labelNames, c.getKey(), v.count));
-      samples.add(new MetricFamilySamples.Sample(fullname + "_sum", labelNames, c.getKey(), v.sum));
+
+    @Override
+    public List<MetricFamilySamples> collect() {
+        List<MetricFamilySamples.Sample> samples = new ArrayList<MetricFamilySamples.Sample>();
+        for (Map.Entry<List<String>, Child> c : children.entrySet()) {
+            Child.Value v = c.getValue().get();
+
+            samples.add(new MetricFamilySamples.Sample(fullname + "_count", labelNames, c.getKey(), v.count));
+            samples.add(new MetricFamilySamples.Sample(fullname + "_sum", labelNames, c.getKey(), v.sum));
+
+
+            // If the quantiles is a desired calculation then add the quantile calculations to the Summary.
+            if (this.shouldCalculateQuantiles()) {
+                // Since `labelNames` is declared final, we need to create a new list that includes our "quantile"
+                // label.
+                double[] data = c.getValue().reservoir.getSnapshot();
+                Arrays.sort(data);
+
+                List<String> labelNamesWithQuantile = new ArrayList<String>(labelNames);
+                labelNamesWithQuantile.add("quantile");
```

Comment:
The quantile label needs to be banned up in Builder, see how it's done in Histogram.